scene/main/viewport.cpp

```diff
@@ -1497,24 +1506,32 @@ String Viewport::_gui_get_tooltip(Control *p_control, const Vector2 &p_pos, Cont
 			break;
 		}
 
+		// Transform cursor pos for parent control.
+		pos = p_control->get_transform().xform(pos);
+
 		p_control = p_control->get_parent_control();
 	}
 
 	return tooltip;
 }
 
 void Viewport::_gui_show_tooltip() {
-	if (!gui.tooltip) {
+	if (!gui.tooltip_control) {
 		return;
 	}
 
+	// Get the Control under cursor and the relevant tooltip text, if any.
 	Control *which = nullptr;
-	String tooltip = _gui_get_tooltip(gui.tooltip, gui.tooltip->get_global_transform().xform_inv(gui.last_mouse_pos), &which);
-	tooltip = tooltip.strip_edges();
-	if (tooltip.length() == 0) {
-		return; // bye
+	String tooltip_text = _gui_get_tooltip(
```

Comment:
Very nice clean up on this one, especially the confusing variable names. I wasn't sure when `tooltip` refers to the control and when to the string when I was looking at that code. Since you are making changes to that function, I'd also rename the `which` variable to something more descriptive. Perhaps `tooltip_text_source`? Or `tooltip_text_source_node`, since `tooltip_text_source_control` would be a bit confusing, haha.

Reply:
I thought about renaming `which` but since it's only used in local code (and a helper method) I didn't feel it was so problematic. But here goes, I renamed it to `tooltip_owner`.